.github/workflows/_move_ide_tests.yml

```diff
@@ -0,0 +1,127 @@
+name: Move IDE Tests
+
+on: workflow_call
+
+concurrency:
+  group: move-ide-tests-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  move-auto-formatter-test:
+    name: Move Auto-formatter Test
+    runs-on: [self-hosted]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # Pin v4.1.1
+        with:
+          ref: ${{ github.event.inputs.iota_repo_ref || github.ref }}
+
+      - name: Setup Node
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # pin@v4.0.2
+        with:
+          node-version: "20"
+
+      - name: Install dependencies
+        working-directory: ./external-crates/move/crates/move-analyzer/prettier-plugin
+        run: npm install && npm i web-tree-sitter
+
+      - name: Run npm test
+        working-directory: ./external-crates/move/crates/move-analyzer/prettier-plugin
+        shell: bash
+        run: npm run test
+
+  move-ide-test:
+    name: Move IDE Test
+    runs-on: [self-hosted]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # Pin v4.1.1
+        with:
+          ref: ${{ github.event.inputs.iota_repo_ref || github.ref }}
+
+      - name: Setup Node
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # pin@v4.0.2
+        with:
+          node-version: "20"
+
+      - name: Install dependencies and start Xvfb (emulate display)
+        working-directory: ./external-crates/move/crates/move-analyzer/editors/code
+        run: |
+          sudo apt install libgtk-3-0 -y
+          sudo apt-get install -y xvfb x11-apps x11-xkb-utils libx11-6 libx11-xcb1
+          set -eux
+          # Start server
+          /usr/bin/Xvfb :99 -screen 0 1024x768x24 &
+          sleep 1
+          ps aux | grep Xvfb --color=always | grep -v grep
+          npm install && npm install --save-dev @types/node @types/semver
+          sudo add-apt-repository ppa:kisak/kisak-mesa -y
+          sudo apt update
+          sudo apt upgrade -y
+
+      - name: Build move-analyzer
+        run: |
+          cargo build --bin move-analyzer
+          mkdir -p ~/.iota/bin
+          cp ./target/debug/move-analyzer* ~/.iota/bin
+
+      - name: Run npm test
+        working-directory: ./external-crates/move/crates/move-analyzer/editors/code
+        shell: bash
+        run: DISPLAY=:99.0 npm run test
```

Comment:
There is one more command that is called [pretest](https://github.com/iotaledger/iota/blob/0948c7b3d4980f0a72860d6b553c51020af1adfd/external-crates/move/crates/move-analyzer/editors/code/package.json#L144). Should we change this line to something like this:
```
DISPLAY=:99.0 npm run pretest && npm run test
```

Reply:
Added in https://github.com/iotaledger/iota/pull/4226/commits/9df0fc70fc9735649edd9706ebdcd657ef181e31